bookingapp-fe/src/ultils/request.ts

```diff
@@ -0,0 +1,90 @@
+import axios from 'axios';
+import { getTokenAsync } from './getToken';
+import { url } from './urlApi';
+import { handleErrorShow, handleSuccessShow } from './ultilsApi';
+
+const instance = axios.create({
+  baseURL: url,
+});
+
+instance.interceptors.request.use(async config => {
+  try {
+    const token = await getTokenAsync();
+    config.headers.Authorization = `Bearer ${token}`;
+    config.headers['ngrok-skip-browser-warning'] = true;
+    return config;
+  } catch (error) {
+    throw error;
+  }
+});
+
+const get = (url: string, params?: any): Promise<any> => {
+  return new Promise((resolve, reject) => {
+    instance
+      .get(url, { params })
+      .then(response => {
+        if (response?.data?.data) {
+          resolve(response.data.data);
+        } else {
+          resolve(null);
+        }
```

Comment:
nếu ko có data thì defined 1 cái error objects có message giống vs bên BE, rồi ghi vô đó message ko có data, hoặc 1 cái key, rồi rejects để ngoài kia xử lý,